Added Two Pointer Method

RaiyanMahin opened the pull request:
<!--
Thank you for sending the PR! We appreciate you spending the time to work on these changes.

Help us understand your motivation by explaining why you decided to make this change.
You can learn more about contributing to NeoAlgo here: https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md

Happy Contributing!

-->

### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?
Yes.

### Description

Find a pair of elements in the sorted array whose sum is equal to a target value. 
Can be solved in by O(N) by using Two Pointers Method

### Checklist

- [x] I've read the contribution guidelines.
- [x] I've checked the issue list before deciding what to submit.
- [x] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests

#2292

RaiyanMahin committed:
add Two Pointer Method

RaiyanMahin committed:
Added Two Pointers Method

RaiyanMahin committed:
Delete README.md

RaiyanMahin committed:
Update README.md

iamrajiv commented:
@RaiyanMahin Please provide the related issue number.

iamrajiv commented on C-Plus-Plus/cp/Two Pointers Method.cpp:
```diff
@@ -0,0 +1,72 @@
+
+#include <bits/stdc++.h>
+using namespace std;
+#define endl "\n"
```
@RaiyanMahin Avoid using macros it decreases the code readability.

iamrajiv reviewed (changes requested):
@RaiyanMahin Rename the file to `Two_Pointers_Method.cpp`. Also, I would suggest adding a naive solution as well in the same file.

RaiyanMahin committed:
Delete Two Pointers Method.cpp

RaiyanMahin committed:
Added Two Pointers Method

RaiyanMahin commented:
@iamrajiv I have added the issue number , rename the file as you told and added the naive solution as well. I also remove the macros from my code.
Kindly review.

RaiyanMahin committed:
Update README.md

RaiyanMahin committed:
Added Sliding Window Technique

RaiyanMahin committed:
Added Sliding Window Technique

RaiyanMahin committed:
Delete Calculate_max_sum_of_k_consecutive_elements.cpp

RaiyanMahin committed:
Delete Calculate_max_sum_of_k_consecutive_elements.cpp

RaiyanMahin commented:
@iamrajiv  Sir , I want to make my 2nd pull request. But can't do it.
can you please tell me how can i do it?

iamrajiv commented:
@RaiyanMahin Whenever you make PR try to make a new branch for every PR so that your main branch did not mess with other branches. Read more about it here https://docs.github.com/en/github/collaborating-with-issues-and-pull-requests/about-pull-requests. Also, make sure you update your fork repository regularly so that you get the latest changes from the original repository. Read more about it here https://stackoverflow.com/questions/7244321/how-do-i-update-a-github-forked-repository.

RaiyanMahin committed:
Added Sliding WIndow Technique

RaiyanMahin committed:
Delete Calculate_max_sum_of_k_consecutive_elements.cpp

RaiyanMahin commented:
@iamrajiv Thanks. Still trying to do.

RaiyanMahin commented:
@HarshCasper , @atarax665 Kindly review :)

HarshCasper commented on C-Plus-Plus/cp/Two_Pointers_Method.cpp:
```diff
@@ -0,0 +1,127 @@
+Two pointers Method is a method where one pointer starts from beginning and other from the end and they proceed towards each other simultaneously
+
+*/
+#include <bits/stdc++.h>
```
Why has a second code been added here? Do you understand that this would mess up badly?

HarshCasper commented on C-Plus-Plus/cp/Two_Pointers_Method.cpp:
```diff
@@ -0,0 +1,127 @@
+
+
+
+
```
Remove extra lines of code

HarshCasper reviewed (changes requested):
Why are two codes pushed in the same file?

RaiyanMahin committed:
Update Two_Pointers_Method.cpp

RaiyanMahin commented:
@HarshCasper , previously @iamrajiv told me to add the naive solution in the same file, So
i added it. 
Now, i updated it and keep the most optimal solution only and remove the extra lines of code.
